### DISCUSSION TITLE
Version 2.0 poll (Planning & Discussion)

### DISCUSSION THREAD

**Discussion by Jamminroot** [Jamminroot/intag#29] [Polls]
Hey everyone!

I've been thinking about a few features I'd love to add to the app, but they don't quite fit with the current setup/UI. That got me thinking about and planning for version 2.0.

I've been looking at a few different languages/frameworks that might work well for the new version, but some of the design principles we used in v1 might not mesh with them as smoothly. For instance, we could swap the single-file executable for a nicer-looking interface, or we could further simplify the UI to shrink the executable size. Another option is to consider a background (always-in-memory) app—though it's not my first choice, it could be something worth exploring for some of you.

With that said, I'd love to get your thoughts on the design principles (not features) you feel are most important for this next version. This will help guide me in choosing the right framework for us.

Feel free to drop your thoughts in the poll or share any extra opinions in the comments!

Note that `always-in-memory` vs `use as little RAM\CPU as possible` is a different vote which is not yet in poll, but can be discussed here too.

**Comment by Jamminroot** [comment]
Personally, I'm leaning more towards improving the looks, even if it means moving away from the single-file executable. Sure, it might feel less like a straightforward tool and more like a full-fledged app, but it would give us way more room for UI customization. Plus, the performance hit isn't that big of a deal.

**Reply by trACEroam** [reply]
looks sometimes allude to ease of use

**Comment by Jamminroot** [comment]
Just a quick update - it's not abandoned, but I had some stuff going on, so couldn't really pay much attention to this tool. Hopefully I'll get to it soon

**Reply by Jamminroot** [reply]
Folder processing logic is ready.
Next up is UI

**Reply by Jamminroot** [reply]
A subject to change, as I am not happy with unaligned grid of tag pills, and would prefer a fixed set of columns
![image](https://github.com/user-attachments/assets/a4599e79-3bfe-462d-b3fd-c75622ed7441)

**Comment by Oxika95** [comment]
Hey been using this for a long time just stopping by to add a feature request for version 2 if possible. Would it be possible to use the system to add meta data to other unused columns? I use it to manage job folders at work so sorting isnt a nightmare but would love to be able to add thing like address or crossroads like in some of the columns here:
<img width="1684" height="1279" alt="image" src="https://github.com/user-attachments/assets/14ac7e3f-aeef-4f8e-877c-74ab100fbdc0" />

**Reply by Jamminroot** [reply]
Hmm... I won't give any promises, as right now I don't see how to fit it into existing functionality\UX, but that definitely sounds like a good improvement.

**Comment by Jamminroot** [comment]
The bad news: I ditched the "v2". Not going to work towards this implementation, as I am extremely unhappy with what I was getting, and problems that final product would have no matter what I try.

The good news: v3 is about to be finished.
I am not going to publish the code for it, but this repository would still be the main point to post issues\discuss\etc.

Also, I am publishing new version on Microsoft Store, and a bunch of features are to be bundled:

1) C++ - it will work even faster, and take even less space
2) Similar interface (maybe slightly different, but no major differences)
3) New metadata edits
4) Windows 11 modern context menu (no need to "show more options")

Windows Store version will be paid (maybe someone will support my work). However, I am still going to publish app in release page here.

**Reply by Jamminroot** [reply]
Well, actually a bit more is going to be shoved in to the new version :D Hope to finish in a couple days now

**Reply by Oxika95** [reply]
will the new version still have CLI support?

**Reply by Jamminroot** [reply]
Yes, full CLI support planned. In further version I will also add other metadata support to CLI mode (for now, it's only tags in CLI mode)

**Reply by Oxika95** [reply]
Awesome! I am looking forward to version 3!

**Reply by Jamminroot** [reply]
Decided to stick to v2 btw, as the flutter version I originally aimed for is not published.

As for the C++ version, it's in the releases page here ([as a pre-release](https://github.com/Jamminroot/intag/releases/tag/2.0.0.0))

Not quite sure msix would work as intended, but zip with .exe is also included.

Currently struggling through MS Store publishing

**Comment by Oxika95** [comment]
<img width="486" height="364" alt="image" src="https://github.com/user-attachments/assets/458d72a1-358e-4663-bc6a-cd09e4ee400e" />

I like the new menu, will there be an option to add additional metadata fields? Street and other street would be very useful for adding crossroads

**Reply by Jamminroot** [reply]
Why not, sure, I can add that

**Comment by Oxika95** [comment]
Awesome, thank you so much!

**Reply by Jamminroot** [reply]
Btw, when you install via MSIX (if that's what you did):
1) Do you get modern context menu to work?
2) Do you have to do acrobatics with signature cheks\unsigned packages etc?

**Reply by Oxika95** [reply]
Actually only used within a test folder I have been putting off adopting the new one due to being worried something might break in my batch scripts. Do i need to uninstall and remove the path variable for the old one first?

**Reply by Jamminroot** [reply]
Yes, you need to update both. Good catch btw, didn't think about a proper migration and maybe an uninstall from the menu. Will do that somewhere on the weekend.

If you could test the MSIX - that'd be helpful, too. Still wondering if that would allow modern context menu (the one without "Show more options") work. 

Otherwise, the app is on the store.

**Reply by Oxika95** [reply]
I don't actually use the modern context menu. I hated it so much I went and got a context menu replacement to adapt back to the older menu. Nielsoft Shell I think it's called.
